### PR TITLE
Delete CMake infrastructure for crossgen1 and remove dead code directories

### PR DESCRIPTION
**Pull request by jkoritzinsky**
I've split this into individual commits that should each build successfully and can be reviewed either individually or all together.

Remove all CMake targets that were only used to compose crossgen1.

Delete any folders where the files within were only used to compile crossgen1-only CMake targets.

Delete all CMake code based on the `CROSSGEN_COMPONENT` CMake target property.

The following defines are now never defined and all blocks guarded by them can be considered dead code. We can now

- [ ] Remove code guarded by CROSSGEN_COMPILE
- [ ] Remove code guarded by FEATURE_NATIVE_IMAGE_GENERATION
- [ ] Remove code guarded by FEATURE_READYTORUN_COMPILER (outside of the coreclr/jit directory. It is still used there)
- [ ] Remove code guarded by FEATURE_PREJIT
- [ ] Remove code guarded by FEATURE_NGEN_RELOCS_OPTIMIZATIONS

Contributes to https://github.com/dotnet/runtime/issues/54129 and https://github.com/dotnet/runtime/issues/53007

cc: @dotnet/crossgen-contrib @AaronRobinsonMSFT @elinor-fung @jkotas @agocke

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by trylek — APPROVED** [review]
As my former lead Slava used to say, red is always good. Thank you!

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/coreclr/tools/crossgen/compare.bat [hunk 1]
```diff
@@ -1,168 +0,0 @@
-::mscorlib
```
Comment (am11) [inline src/coreclr/tools/crossgen/compare.bat:1]: nit: also found a few `crossgen1` hits in few other places:
```sh
$ git grep -li crossgen1 :/

docs/design/coreclr/botr/vectors-and-intrinsics.md
src/coreclr/crossgen-corelib.proj
src/coreclr/tools/aot/ILCompiler.Diagnostics/PerfMapWriter.cs
src/coreclr/tools/aot/ILCompiler.ReadyToRun/ObjectWriter/SymbolFileBuilder.cs
src/coreclr/tools/r2rtest/Crossgen2Runner.cs
src/coreclr/tools/r2rtest/CrossgenRunner.cs
src/tasks/Crossgen2Tasks/PrepareForReadyToRunCompilation.cs
src/tasks/Crossgen2Tasks/RunReadyToRunCompiler.cs
src/tests/readytorun/crossboundarylayout/crossboundarytests.cmd
```

Comment (jkoritzinsky) [reply src/coreclr/tools/crossgen/compare.bat:1]: I think we can remove those occurrences in a future PR. I want to keep this one focused on the native build for simplicity.

Comment (trylek) [reply src/coreclr/tools/crossgen/compare.bat:1]: I suspect we also have some residua in the test build infra, around files like CLRTest.Crossgen.targets, R2RTest and such. I'll float their cleanup as one of the ideas for the Quality week next week.

**Comment by MichalStrehovsky** [comment]
Cc @dotnet/samsung crossgen1 is getting removed here, as indicated before - note this is targeting main branch (.NET 7). .NET 6 got snapped into the release/6.0 branch.

**Review by elinor-fung — APPROVED** [review]

**Comment by davidwrighton** [comment]
Could you also remove the references to FEATURE_PREJIT from clrdefinitions.cmake? That code is also now completely dead as of crossgen no longer being in our build.

**Comment by trylek** [comment]
I cannot resist sharing this small reminder regarding our recent journey. Thanks so much to everyone involved!

<pre>
From: Zach Montoya <zamont@microsoft.com> 
Sent: Tuesday, May 1, 2018 10:02 PM
To: Jan Kotas <jkotas@microsoft.com>
Cc: Simon Nattress <simonn@microsoft.com>; Tomas Rylek <trylek@microsoft.com>
Subject: Ready To Run Chat / Brain Dump

Hi Jan,

Would you mind taking an hour sometime this week to give me (and possibly Simon & Tomas if interested) a high-level talk on Ready to Run? I have no experience with Ready to Run at the moment so I’d love to get a brain dump from you so I have an improved understanding of it, especially since Sergiy’s new intern will be arriving on Monday and I’ll be coaching her through the internship.

If that’s fine with you, then Simon & Tomas let me know as well if you would be interested so we can coordinate a good meeting time.

Thanks,
Zach
</pre>

**Review by mangod9 — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Test failures are known and linked. Only build failure is the feed instability. I'm going to merge.

**Comment by jkoritzinsky** [comment]
This removal cut an execution of `./src/coreclr/build-runtime.cmd` from 5 minutes 30 seconds with all cores focusing on compile to 4 minutes 52 seconds while one core was being pegged by my antivirus (out of my control). So that's a reduction of about 7 CPU minutes if everything were on a single thread.

**Comment by am11** [comment]
Thanks! Great impactful cleanup indeed! 8-)

**Pull request merged**